drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/index/RuleBuilder.java

```diff
@@ -59,14 +62,25 @@ private void resolvePatterns() {
             final Pattern pattern = resolvePattern( rule,
                                                     pattern52 );
 
-            new ConditionsBuilder( index,
-                                   model,
-                                   row,
-                                   utils,
-                                   pattern52,
-                                   pattern ).buildConditions();
+            new FieldConditionsBuilder( index,
+                                        model,
+                                        rule,
+                                        row,
+                                        utils,
+                                        pattern ).buildConditions( pattern52.getChildColumns() );
+        }
+
+        for ( final CompositeColumn<? extends BaseColumn> column : model.getConditions() ) {
```

Comment:
You could possibly wrap the iterator over ```getPatterns()``` and ```getConditions()``` into a single loop and use ```instanceof Pattern52``` and ```instanceof BRLConditionColumn``` inside the same loop.